File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = [];
    const map1 = arr1.map(value => {
        return value.toString(2).padStart(n, 0);
    })
    
    const map2 = arr2.map(value => {
        return value.toString(2).padStart(n, 0);
    })
    for(let i = 0; i < n; i++){
        let str = ''
        for(let j = 0; j < n; j++){
            if(!(map1[i][j] === '0' && map2[i][j] === '0')){
                str += '#';
            }else{
                str += ' '
            }
        }
        answer.push(str);
    }
    
    return answer;
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    grid1 = [str(bin(v)[2:]).zfill(n) for v in arr1]
    grid2 = [str(bin(v)[2:]).zfill(n) for v in arr2]
    for g1, g2 in zip(grid1, grid2):
        temp = ''
        for v1, v2 in zip(g1, g2):
            if v1 == '0' and v2 == '0':
                temp += ' '
            else:
                temp += '#'
        answer.append(temp)
            
    return answer